[DOCS-9487] NDM Device onboarding updates (#27598)
DataDog/documentation: content/en/network_monitoring/devices/guide/device_profiles.md

5. Select the metric type, scale factor, and extract value (regex pattern). 
See [advanced options for scalar metrics](?tab=manual#scalar-metrics) and [advanced options for tabular metrics](?tab=manual#tabular-metrics) for more information.
6. Click **Create** to save the metric.
7. This returns you to the define metrics screen where you can see the new metric that was added.

**Note**: To avoid a validation error for tabular metrics, at least one metric tag must be added on the Define metrics screen. 

{{< img src="/network_device_monitoring/profile_onboarding/add_metrics_manually.mp4" alt="Video showing the add metrics modal, adding a new metric with the manual method and returning to define metrics step." video=true >}}

{{% collapse-content title="Advanced options scalar" level="h4" expanded=false %}}

#### Scalar metrics

[Metric Type][11]
: One of `gauge`, `rate`, `monotonic_count`, or `monotonic_count_and_rate`. 

[Scale Factor][12]
: Before being transmitted to Datadog, the extracted value is multiplied by this factor.

[Extract Value][7]

> : This is the same as the [advanced options](?tab=manual#global-tags-advanced-options) for global tags.

Author: fixed

  3. Provide your device profile a name and description (optional).
  4. Select the `SysObjectID`. This is what is used to match network devices to the device profiles that define what is collected and monitored from each device. 

     {{< img src="/network_device_monitoring/profile_onboarding/Sys_object_ID_Field_2.png" alt="The Network Device profile creation page showing the Sys Object ID Dropdown" style="width:60%;">}}

### Step 2: Profile inheritance 

Use profile inheritance to adopt configurations such as metadata, metrics, and tags. This simplifies scaling your device profiles and allows you to build on existing ones. Datadog automatically includes some inherited profiles, labeled as `_base.yaml`, which are recommended **not** to be removed. Reference the [Supported Device Profiles][16] for the full list of inherited profiles.

1. Keep the Datadog `_base.yaml` profile, and optionally, select one or more profiles to inherit. The relevant fields appear on the right under Inherited Profiles, with an `Inherited` tag next to any inherited metrics, tags, or metadata:

>    {{< img src="/network_device_monitoring/profile_onboarding/profile_inheritance.png" alt="The Network Device profile creation page showing the Profile inheritance section." style="width:100%;">}}

Author: Thank you for noting this! I'll update this

    **Note**: Changes made to the parent profile are automatically applied to the child profiles. If you need to adjust any inherited attributes in the child profiles, such as metrics, tags, or metadata, you must modify the parent profile.

### Step 3: Select reference devices

Use reference devices to select which devices you want to gather OIDs for your chosen device models. The **reference devices** field is pre-selected based on the `SysObjectID` that was specified in the [profile details](#step-1-profile-details).

1. Keep the current reference device selection to perform a device scan. Additionally, you can add more devices or change the current selection.

2. Optionally, click **Proceed Manually** to proceed without performing a scan.

  {{< img src="/network_device_monitoring/profile_onboarding/reference_devices.png" alt="The Network Device profile creation page showing the Reference device section." style="width:100%;">}}

2. Click **Scan Devices** to proceed to Step 4, which initiates the scan.

### Step 4: Scan reference devices

> This step scans your devices to discover their available metrics. Running a scan allows you to view all available metrics for your device, making it easier to fill in metrics, metadata, and tags. The scan performs an SNMP walk on your devices using [Datadog Remote Configuration][14]. 

Author: ok thanks @NouhaManai96 ! I will leave this comment open until the UI is ready.

> This step scans your devices to discover their available metrics. Running a scan allows you to view all available metrics for your device, making it easier to fill in metrics, metadata, and tags. The scan performs an SNMP walk on your devices using [Datadog Remote Configuration][14]. 

Author: Confirmed this is not a blocker for now so resolving this comment and can iterate on this laster.

  {{< img src="/network_device_monitoring/profile_onboarding/define_metadata.png" alt="The Network Device profile creation page showing the define metrics section." style="width:80%;">}}

  1. Click the pencil icon to edit and modify any of the default metadata fields.
  
  2. Metadata functionality is available and displayed on the [Network Device Monitoring (NDM)][15] page as searchable facets, and on the side panel of a selected device:

     {{< img src="/network_device_monitoring/profile_onboarding/device_metadata_2.png" alt="The NDM side panel page profile, highlighting the metadata sections." style="width:100%;">}}

### Step 6: Define metrics

Metrics can be added either from a device scan or by manually creating a new metric for the profile. Inherited metrics are highlighted in purple with the `Inherited` tag.

{{< tabs >}}
{{% tab "Device scan (recommended)" %}}

1. To define a metric using the **Device Scan** option, click **Add Metrics**. This opens a modal displaying all available metrics for the device.
2. Hover over metrics to see units and descriptions for easier selection.
3. Select the metrics you wish to add, then click **Add Metric**.

> 4. This returns you to the define metrics screen where you can see the new metric that was added.

Author: Ack, will add this, however, I'm wondering would they only run into this situation if they did not select any inherited profiles ?

> 4. This returns you to the define metrics screen where you can see the new metric that was added.

Reviewer: If they selected inherited profiles then the inherited metrics will be already paired with the tags.
This happens when adding a new table metric (either manually or from the scan data)

   {{< img src="/network_device_monitoring/profile_onboarding/define_metrics_2.mp4" alt="Video showing the add metrics modal, adding a new metric and returning to define metrics step." video=true >}}

{{% /tab %}}

{{% tab "Manual" %}}

1. To define a metric using the **Manual** option, click **Add Metrics**. This opens a modal displaying all available metrics for the device.
2. Click **Create New Metric** at the top of the modal.
3. Specify the OID (Scalar, or Tabular). 

> 4. Click the dropdown in the search field to add the OID name.

Author: ack, added

    **Note**: Changes made to the parent profile are automatically applied to the child profiles. If you need to adjust any inherited attributes in the child profiles, such as metrics, tags, or metadata, you must modify the parent profile.

### Step 3: Select reference devices

Use reference devices to select which devices you want to gather OIDs for your chosen device models. The **reference devices** field is pre-selected based on the `SysObjectID` that was specified in the [profile details](#step-1-profile-details).

1. Keep the current reference device selection to perform a device scan. Additionally, you can add more devices or change the current selection.

2. Optionally, click **Proceed Manually** to proceed without performing a scan.

  {{< img src="/network_device_monitoring/profile_onboarding/reference_devices.png" alt="The Network Device profile creation page showing the Reference device section." style="width:100%;">}}

2. Click **Scan Devices** to proceed to Step 4, which initiates the scan.

> 

Reviewer 2: I would move the primary instruction above the image, i almost missed it when reading

    </details>

> {{% /collapse-content %}}

Author: thanks, fixed!

{{% /tab %}}
{{< /tabs >}}

### Step 7: Global Tags

Add global tags to ensure the metadata, metrics, and global tags are applied to all matching devices. Global tags can be added either from a device scan or by manually creating a new tag for the profile. Additionally, the `Inherited` tag will appear next to any global tags inherited from this profile. 

{{< tabs >}}
{{% tab "Device scan (recommended)" %}}

1. To define a global tag using the **Device Scan** option, click **+ Add Tags**. This opens a modal displaying all available tags for the device.
2. Select one or more tags you wish to add to the device profile, then click **Add Tag**.

> 3. This returns you to the define global tags screen where you can see the new tag that was added.

Reviewer: At this step, customers can also edit the tag to add any modification (similarly to the manual flow)
```suggestion
3. This returns you to the define global tags screen where you can see and edit the new tag that was added.
```

   {{< img src="/network_device_monitoring/profile_onboarding/add_global_tag.mp4" alt="Video showing the add global tags modal, adding a new tag and returning to global tags step." video=true >}}

{{% /tab %}}

{{% tab "Manual" %}}

1. To define a global tag using the **Manual** option, click **+ Add Tags**. This opens a modal displaying all available tags for the device.
2. Click **Create New Tag** at the top of the modal.
3. Select the dropdown in the search field to add the OID name.

> 4. Click the **Modification** dropdown to add a modification. See advanced [options](?tab=manual#global-tags).

Reviewer: ```suggestion
4. Click the **Modification** dropdown to add a modification. See [advanced options](?tab=manual#global-tags).
```

| Extract Value   | A regular expression used to [extract][7] the tag value from the SNMP value provided by the device. |
| Mapping         | This is described [here][8].   

[5]: https://datadoghq.dev/integrations-core/tutorials/snmp/profile-format/#format-mac_address
[6]: https://datadoghq.dev/integrations-core/tutorials/snmp/profile-format/#format-ip_address
[7]: https://datadoghq.dev/integrations-core/tutorials/snmp/profile-format/#extract_value
[8]: https://datadoghq.dev/integrations-core/tutorials/snmp/profile-format/#mapping-index-to-tag-string-value

{{% /collapse-content %}}

{{% /tab %}}
{{< /tabs >}}

### Apply a profile to created devices

{{< tabs >}}
{{% tab "Automatic (recommended)" %}}

After applying your configuration options to your device profile, click **Save and Sync Agents** to automatically apply this profile to all NDM agents. The configurations are applied to your devices with [Remote Configuration][14].

{{< img src="/network_device_monitoring/profile_onboarding/save_sync_agents.png" alt="The Network Device profile page showing the final step to save and sync agents" style="width:100%;">}}

[14]: /agent/remote_config

{{% /tab %}}

{{% tab "Manual" %}}

> 1. After you save a profile as a draft, navigate back to the [profile home page][4], select the **Draft Profiles** tab, then select the **Download All Profiles** option. This allows you to download the `.zip` bundle which contains the `yaml` files for the profiles you created. 

Reviewer: The draft mode is used as a way to facilitate the profile creation process for customers since it can be a long flow, they can use the draft mode when the profile is still in progress (ie not ready to be synced and applied to agents).
Draft profiles are excluded from any real operation like the sync to agents or the download.
So any profile marked as draft **won't be** downloaded.
Also that the download option means that **all** profiles will be downloaded in a zip together. Customers can't chose what profiles to download no matter what tab they select.

2. Place the `yaml` files in the [profile directory][13] on each of the relevant installed Agents.
3. Restart the Datadog Agent.
4. To ensure the profiles you created are accurate, confirm that NDM is receiving metrics from the matched devices as expected.

{{< img src="/network_device_monitoring/profile_onboarding/profile_download.png" alt="The Network Device profile main page highlighting the Download All Profiles option" style="width:100%;">}}

[4]: https://app.datadoghq.com/devices/profiles
[13]: https://github.com/DataDog/integrations-core/tree/master/snmp/datadog_checks/snmp/data/profiles

{{% /tab %}}
{{< /tabs >}}

## Inventory page

> Navigate to [Infrastructure > Network Devices > Configuration][1] to view the device profile [Inventory][4] page. This is where you can see a snapshot of the device profiles you created using the [Device Profile Setup](#setup) below. 

Reviewer: The view shows also the datadog OOTB profiles (which were not created through the UI)

> Navigate to [Infrastructure > Network Devices > Configuration][1] to view the device profile [Inventory][4] page. This is where you can see a snapshot of the device profiles you created using the [Device Profile Setup](#setup) below. 

Reviewer: ```suggestion
Navigate to [Infrastructure > Network Devices > Configuration][1] to view the device profile [Inventory][4] page. This is where you can see a snapshot of the device profiles you created using the [Device Profile Setup](#setup) above. 
```

  {{< img src="/network_device_monitoring/profile_onboarding/device_status.png" alt="Screenshot of a device profile showing the draft state status" style="width:50%;">}}

- **Filters**: The filters include the following options:
  - Custom Profiles - User created device profiles.
  - Created by Datadog - Datadog out-of-the-box profiles that can be viewed and used to build your own custom profile.
  - Draft Profiles - Devices profiles in that are in draft mode.

  {{< img src="/network_device_monitoring/profile_onboarding/device_filters.png" alt="Screenshot of the device profile inventory page showing the filter options" style="width:60%;">}}

- **Create new profile and Download**: The **+ Create New Profile** button opens the profile creation form which allows you to [build a new device profile.](#build-device-profiles) Clicking the download button will generate and download a `.zip` bundle which contains the `yaml` files for the profiles you created. <br></br>

  {{< img src="/network_device_monitoring/profile_onboarding/create_profile_download.png" alt="Screenshot of the device profile inventory page showing the download and create a new profile buttons" style="width:50%;">}}

> - **Kebab menu**: Clicking the kebab menu to the right of a profile allows you to edit, clone, or delete the profile (for custom or draft profiles only). You can also navigate to the **View related devices** on the NDM page, filtered to the device the profile is applied to.<br></br>

Reviewer: the draft profiles are custom profiles
```suggestion
- **Kebab menu**: Clicking the kebab menu to the right of a profile allows you to edit, clone, or delete the profile (for custom profiles only). You can also navigate to the **View related devices** on the NDM page, filtered to the device the profile is applied to.<br></br>
```

> - **Kebab menu**: Clicking the kebab menu to the right of a profile allows you to edit, clone, or delete the profile (for custom or draft profiles only). You can also navigate to the **View related devices** on the NDM page, filtered to the device the profile is applied to.<br></br>

Reviewer: ```suggestion
- **Kebab menu**: Clicking the kebab menu to the right of a profile allows you to edit, clone, or delete the profile (for custom or draft profiles only). You can also navigate to the **View related devices** on the NDM page, filtered to the device(s) the profile is applied to.<br></br>
```

Author: fixed

  {{< img src="/network_device_monitoring/profile_onboarding/create_profile_download.png" alt="Screenshot of the device profile inventory page showing the download and create a new profile buttons" style="width:50%;">}}

- **Kebab menu**: Clicking the kebab menu to the right of a profile allows you to edit, clone, or delete the profile (for custom profiles only). You can also navigate to the **View related devices** on the NDM page, filtered to the device(s) the profile is applied to.<br></br>

  {{< img src="/network_device_monitoring/profile_onboarding/device_kebab_menu.png" alt="Screenshot of the device profile inventory page showing the kebab menu on the right hand side" style="width:40%;">}}
                                                  
## Troubleshooting

- What is a profile?
- What is a device scan?
- Why do I see no matching devices? 
- What happens if I don't have remote configuration enabled on my collectors? need screenshot of error state that directs them to Fleet automation to upgrade to the most recent Datadog Agent or run scan manually.
- Why would a device not be scanned?

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

> [1]: https://app.datadoghq.com/devices/enrichment

Reviewer 3: ```suggestion
[1]: https://app.datadoghq.com/devices/profiles
```

Author: good catch fixed!

  {{< img src="/network_device_monitoring/profile_onboarding/device_status.png" alt="Screenshot of a device profile showing the draft state status" style="width:50%;">}}

- **Filters**: The filters include the following options:
  - Custom Profiles - User created device profiles.
  - Created by Datadog - Datadog out-of-the-box profiles that can be viewed and used to build your own custom profile.
  - Draft Profiles - Devices profiles in that are in draft mode.

  {{< img src="/network_device_monitoring/profile_onboarding/device_filters.png" alt="Screenshot of the device profile inventory page showing the filter options" style="width:60%;">}}

- **Create new profile and Download**: The **+ Create New Profile** button opens the profile creation form which allows you to [build a new device profile.](#build-device-profiles) Clicking the download button generates and download a `.zip` bundle which contains the `yaml` files for the profiles you created. <br></br>

  {{< img src="/network_device_monitoring/profile_onboarding/create_profile_download.png" alt="Screenshot of the device profile inventory page showing the download and create a new profile buttons" style="width:50%;">}}

> - **Kebab menu**: Clicking the kebab menu to the right of a profile allows you to edit, clone, or delete the profile (for custom profiles only). You can also navigate to the **View related devices** on the NDM page, filtered to the device(s) the profile is applied to.<br></br>

Reviewer 4: ```suggestion
- **Kebab menu**: Clicking the kebab menu to the right of a profile allows you to edit, clone, or delete the profile (for custom profiles only). You can also navigate to **View related devices** on the NDM page, filtered to the device(s) the profile is applied to.<br></br>
```

  {{< img src="/network_device_monitoring/profile_onboarding/device_kebab_menu.png" alt="Screenshot of the device profile inventory page showing the kebab menu on the right hand side" style="width:40%;">}}
                                                  
## Troubleshooting

### What is a profile?
* A profile is a configuration file that defines the metadata, metrics, and tags to collect from a device. See [metadata definition][17] for more information.

### What is a device scan?
* A device scan conducts a complete SNMP walk of the device, collecting all available data and forwarding it to Datadog for display in the UI. This process helps you identify the OIDs available on the device and add them to the profile for monitoring.

### Why would I see no matching devices? 
If no matching devices are found, it may be due to the following reasons:  
  * **The profile is in Draft mode**:

>     * Draft profiles are not applied to the Agent. To start monitoring devices with your profile you must sync it to the Agent(s). This can be done by opening the profile and clicking on the [**Save & Sync Agents**](#apply-a-profile-to-created-devices) button.  

Reviewer 4: ```suggestion
    * Draft profiles are not applied to the Agent. To start monitoring devices with your profile, you must sync it to the Agent(s). This can be done by opening the profile and clicking on the [**Save & Sync Agents**](#apply-a-profile-to-created-devices) button.  
```

  * **Multiple profiles have the same SysObjectID(s)**:  
    * Profiles are matched to devices using their SysObjectID. If multiple profiles share the same SysObjectID, it can cause matching conflicts at the Agent level. Ensure that each [SysObjectID](#step-1-profile-details) is assigned to only one profile.

### Why would a device not be scanned?

* The device scan may take up to 10 minutes to complete. You can monitor the scan's progress in the UI. If errors occur, try restarting the scan or selecting a different [reference device](#step-3-select-reference-devices).

### What if I don't have Remote Configuration enabled on my collectors? 

* If you are using an Agent version earlier than `7.47.0` and do not already have [Remote Configuration][18] manually enabled on your hosts, you will not be able to trigger device scans or sync profiles to the Agents through the UI. However, you can perform these steps manually. <br /><br>

   To scan a device, follow the instructions in the UI: <br /><br>

   {{< img src="/network_device_monitoring/profile_onboarding/remote_configuration.png" alt="Screenshot of the " style="width:80%;">}}

>    Or, to apply the profiles to your Agent(s) manually:  

Reviewer 4: ```suggestion
   Or, to apply the profiles to your Agents manually:  
```

     1. Save the profile.  
     2. Click on the download button to save a zip file of all your profiles.  

>      3. Upload the zip file to your Agent(s) by following the instructions in the [manually apply a profile to created devices][19] section.

Reviewer 4: ```suggestion
     3. Upload the zip file to your Agents by following the instructions in the [manually apply a profile to created devices][19] section.
```

Datadog strongly recommends enabling Remote Configuration to ensure a seamless, UI-based experience and to minimize unnecessary interactions with the Agent.

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

> [1]: https://app.datadoghq.com/devices/profiles

Reviewer 4: Links 1 and 4 are the same.

Author: fixed
